health/health.go

```diff
@@ -0,0 +1,40 @@
+package health
+
+import (
+	"errors"
+	"sync"
+)
+
+var Reporter = &reporter{
+	ch:     make(chan error, 1),
+	errors: map[string]error{},
+	mut:    sync.Mutex{},
+}
+
+type reporter struct {
+	ch     chan error
+	errors map[string]error
+	mut    sync.Mutex
+}
+
+func (r *reporter) NewHealth(component string, err error) {
+	r.mut.Lock()
+	defer r.mut.Unlock()
+	r.errors[component] = err
+	list := make([]error, 0, len(r.errors))
+	for _, err := range r.errors {
+		if err != nil {
+			list = append(list, err)
+		}
+	}
+
+	r.ch <- errors.Join(list...)
```

Comment:
JFYI this doesn't work with multiple-line errors, as each newline char will be interpreted as new entry. Here's a reproducer: Running the tests, I've noticed that one error message is separated from another one with the `\n` newline character. For example, the current output of tests is 4 errors:

```
component1 went wrong
component1 went wrong
component2 went wrong
component1 went wrong
```

but if I modify (not adding a new test, just modifying a already existing test-case) such as:

```diff
- err:       errors.New("component2 went wrong"),
+ err:       errors.New("component2 went wrong\ncomponent2 went wrong again"),
 ```

then running the tests it produces 5 messages:

```
component1 went wrong
component1 went wrong
component2 went wrong
component2 went wrong again
component1 went wrong
```